### ISSUE TITLE
🐛 (BETA) Infinite loop when processing a tiny code snippet

### ISSUE THREAD

**Issue by jpike88** [biomejs/biome#5985]
### Environment information

```block
CLI:
  Version:                      2.0.0-beta.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.0.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.3.0

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            false
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0
```

### What happened?

This code by itself in a typescript file is all it took to make biome go nuts:

```
import moment from 'moment';
import { Moment } from 'moment';
```

The command I used:

`node_modules/@biomejs/biome/bin/biome check --write THE_FILE_NAME_GOES_HERE --colors=force`

Granted, it's not pretty code, but at least it should be flagged as rule or something... typescript, angular etc didn't say anything wrong with it.

biome.json:

```
{
	"$schema": "./node_modules/@biomejs/biome/configuration_schema.json",
	"formatter": {
		"enabled": false
	},
	"assist": { "actions": { "source": { "organizeImports": "on" } } },
	"javascript": {
		"formatter": {
			"quoteStyle": "single",
			"trailingCommas": "es5"
		},
		"parser": {
			"unsafeParameterDecoratorsEnabled": true
		}
	},
	"linter": {
		"enabled": true,
		"rules": {
			"a11y": {
				"noNoninteractiveTabindex": "error",
				"noRedundantRoles": "error"
			},
			"complexity": {
				"noBannedTypes": "off",
				"noExtraBooleanCast": "off",
				"useSimplifiedLogicExpression": "off",
				"useOptionalChain": "error",
				"noUselessConstructor": "off",
				"noUselessTypeConstraint": "error",
				"noUselessRename": "error",
				"useLiteralKeys": "error",
				"noForEach": "error",
				"noStaticOnlyClass": "off",
				"useArrowFunction": "off",
				"noThisInStatic": "off",
				"useSimpleNumberKeys": "off",
				"useRegexLiterals": "off"
			},
			"performance": {
				"noDelete": "off"
			},
			"suspicious": {
				"noTemplateCurlyInString": "off",
				"noGlobalIsNan": "off",
				"noDoubleEquals": "error",
				"noRedeclare": "off",
				"noDebugger": "error",
				"noShadowRestrictedNames": "off",
				"noExplicitAny": "off",
				"useValidTypeof": "error",
				"noAsyncPromiseExecutor": "off",
				"noEmptyInterface": "error",
				"noAssignInExpressions": "error",
				"noConsole": {
					"level": "error",
					"options": {
						"allow": ["info", "error", "warn", "time", "timeEnd"]
					}
				},
				"noPrototypeBuiltins": "error",
				"noVar": "error"
			},
			"style": {
				"useConsistentArrayType": {
					"level": "error",
					"options": { "syntax": "shorthand" }
				},
				"useForOf": "error",
				"useImportType": "off",
				"noNamespace": "off",
				"noParameterAssign": "off",
				"useSingleVarDeclarator": "error",
				"useTemplate": "off",
				"noUnusedTemplateLiteral": "error",
				"useConst": "error",
				"useEnumInitializers": "off",
				"useDefaultParameterLast": "off",
				"noInferrableTypes": "off",
				"useLiteralEnumMembers": "error",
				"noUselessElse": "off",
				"noDefaultExport": "error",
				"useAsConstAssertion": "error",
				"useSelfClosingElements": "error",
				"useNumberNamespace": "error",
				"useNamingConvention": {
					"level": "error",
					"options": {
						"strictCase": false
					}
				}
			},
			"correctness": {
				"noConstantCondition": "error",
				"noInvalidConstructorSuper": "error",
				"noUnusedVariables": "error",
				"noUnreachable": "error",
				"noUnsafeFinally": "error",
				"noSwitchDeclarations": "off",
				"noSelfAssign": "off"
			}
		}
	},
	"files": {
		"includes": [
			"**",
			"!**/*.json",
			"!**/*.d.ts",
			"!**/client/src/components/api.ts",
			"!**/seed/src/components/api.ts",
			"!**/tests/src/integration/api.ts",
			"!**/client/src/components/model/TranslationKeys.ts",
			"!**/server/src/components/translations/TranslationKeys.ts",
			"!**/shared/version.ts",
			"!**/Exif.ts",
			"!**/Measurement.ts",
			"!**/ConfigFile.ts",
			"!**/Converter.ts",
			"!**/importBookingUtil.ts",
			"!**/*.js"
		]
	}
}
```

### Expected result

Biome not spin infinitely 

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

edit: seems like this is a general issue anywhere there are dupe import sources.

**Comment by ematipico** [comment]
@jpike88 did you manage to understand **what** is causing the loop? Did you try turning off rules/actions?

**Comment by jpike88** [comment]
Not much luck. Even if I strip the biome.json file down to almost nothing, it still happens. I thought at first it was the organise import action, but seems like something deeper. 

Adding log level debug didn't help much either (I redacted some filepaths):


```
  2025-05-12T16:21:06.809746Z  INFO  Configuration file loaded: Some("/xxxxxx"), diagnostics detected 0
    at crates/biome_cli/src/commands/mod.rs:807 on main

  2025-05-12T16:21:06.809794Z DEBUG  Insert workspace folder /Users/xxxxx
    at crates/biome_service/src/projects.rs:45 on main
    in insert_project

  2025-05-12T16:21:06.810242Z DEBUG  The file has the following feature sets: {Debug: Supported, Assist: Supported, Format: Supported, Search: Supported, Lint: Supported}
    at crates/biome_service/src/workspace.rs:254 on biome::workspace_worker_0
    in with_settings_and_language with path: "/Users/xxxxx"
    in can_handle

  2025-05-12T16:21:06.810286Z DEBUG  The file has the following feature sets: {Assist: Supported, Debug: Supported, Format: Supported, Lint: Supported, Search: Supported}
    at crates/biome_service/src/workspace.rs:254 on biome::workspace_worker_0
    in with_settings_and_language with path: "/Users/xxxx/xxxx.ts"

```

Then it's just stuck at that point, process spins at 99% cpu till I terminate the process

**Comment by ematipico** [comment]
cc @Conaclos

**Comment by Conaclos** [comment]
This is certainly caused by one of the last features I added: import merging.

**Comment by Conaclos** [comment]
I forget to handle default and named imports merging. Then the action emits nothing, leading to an infinite loop.
I am a bit surprised that we have not a limit to avoid an endless loop.
